src/api/procedures/createGroup.ts

```diff
@@ -79,33 +80,26 @@ export async function prepareCreateGroup(
   const rawTicker = stringToTicker(ticker, context);
   const { transactions } = permissionsLikeToPermissions(permissions, context);
 
-  const { data: groups } = await token.permissions.getGroups();
+  const groups = await token.permissions.getGroups();
 
-  const currentTransactionPermissions: TransactionPermissions[] = [];
-
-  await Promise.all(
-    groups.map(async group => {
-      const { transactions: groupTransactions } = await group.getPermissions();
-      if (groupTransactions) {
-        currentTransactionPermissions.push(groupTransactions);
-      }
-    })
-  );
+  const currentGroupPermissions = await P.map(groups, group => group.getPermissions());
 
   if (
-    currentTransactionPermissions.some(transactionPermission =>
-      isEqual(transactionPermission, transactions)
+    currentGroupPermissions.some(({ transactions: transactionPermissions }) =>
+      isEqual(
+        optionize(orderTransactionPermissionsValues)(transactionPermissions),
```

Comment:
The permissions that come from the chain are already ordered. You only need to order the input